plugins/kinetic-dp/fu-kinetic-dp-secure-aux-isp.c

```diff
@@ -0,0 +1,873 @@
+	dev_info->fw_info.customer_project_id = dpcd_buf[12];							/* DPCD 0x515		*/
+	dev_info->fw_info.customer_fw_ver = (guint16)dpcd_buf[6] << 8 | (guint16)dpcd_buf[11];			/* DPCD (0x50F | 0x514)	*/
+	dev_info->chip_type = dpcd_buf[13];									/* DPCD 0x516		*/
+
+	if (KT_FW_STATE_RUN_APP == dev_info->fw_run_state) {
+		dev_info->is_dual_bank_supported = TRUE;
+		dev_info->flash_bank_idx = fu_kinetic_dp_secure_aux_isp_get_flash_bank_idx (connection, error);
+	}
+
+	dev_info->fw_info.boot_code_ver = 0;
+	/* <TODO> Add function to read CMDB information */
+	dev_info->fw_info.std_cmdb_ver = 0;
+	dev_info->fw_info.cmdb_rev = 0;
+
+	return TRUE;
+}
+
+static gboolean
+fu_kinetic_dp_secure_aux_isp_start_isp (FuKineticDpDevice *self,
+					FuFirmware *firmware,
+					const KtDpDevInfo *dev_info,
+					GError **error)
+{
+	FuKineticDpFirmware *firmware_self = FU_KINETIC_DP_FIRMWARE (firmware);
+	gboolean ret = FALSE;
+	gboolean is_app_mode = (KT_FW_STATE_RUN_APP == dev_info->fw_run_state) ? TRUE : FALSE;
+	g_autoptr (GBytes) isp_drv = NULL;
+	g_autoptr (GBytes) app = NULL;
+	const guint8 *payload_data;
+	gsize payload_len;
+	g_autoptr (FuDeviceLocker) locker = NULL;
+	g_autoptr (FuKineticDpConnection) connection = NULL;
+	g_autoptr (FuFirmware) img = NULL;
+
+	connection = fu_kinetic_dp_connection_new (fu_udev_device_get_fd (FU_UDEV_DEVICE (self)));
+
+	isp_procd_size = 0;
+
+	g_debug ("Start secure AUX-ISP [%s]...", fu_kinetic_dp_aux_isp_get_chip_id_str (dev_info->chip_id));
+
+	/* Write MCA OUI */
+	if (!fu_kinetic_dp_secure_aux_isp_write_mca_oui (connection, error))
+		goto SECURE_AUX_ISP_END;
+
+	/* Send ISP driver and execute it */
+	img = fu_firmware_get_image_by_idx (firmware, FU_KT_FW_IMG_IDX_ISP_DRV, error);
+	if (NULL == img)
+		return FALSE;
+
+	isp_drv = fu_firmware_write (img, error);
+	if (isp_drv == NULL)
+		return FALSE;
+
+	payload_data = g_bytes_get_data (isp_drv, &payload_len);
+	if (payload_len) {
+		if (!fu_kinetic_dp_secure_aux_isp_send_isp_drv (connection, is_app_mode, payload_data, payload_len, error))
+			goto SECURE_AUX_ISP_END;
+	}
+
+	/* Enable FW update mode */
+	if (!fu_kinetic_dp_secure_aux_isp_enable_fw_update_mode (firmware_self, connection, error))
+		goto SECURE_AUX_ISP_END;
+
+	/* Send App FW image */
+	img = fu_firmware_get_image_by_idx (firmware, FU_KT_FW_IMG_IDX_APP_FW, error);
+	if (NULL == img)
+		return FALSE;
+
+	app = fu_firmware_write (img, error);
+	if (app == NULL)
+		return FALSE;
+
+	payload_data = g_bytes_get_data (app, &payload_len);
+	if (!fu_kinetic_dp_secure_aux_isp_send_fw_payload (connection, firmware_self, payload_data, payload_len, error))
+		goto SECURE_AUX_ISP_END;
+
+	/* Install FW images */
+	ret = fu_kinetic_dp_secure_aux_isp_install_fw_images (connection, error);
+
+SECURE_AUX_ISP_END:
+	/* Send reset command */
+	fu_kinetic_dp_secure_aux_isp_send_reset_command (connection, error);
```

Comment:
you can't re-use the error here if it's set. Maybe just a g_warning and error_local?